src/layout/commands.rs

```diff
@@ -368,4 +370,26 @@ impl Tree {
         self.0.send_active_to_workspace(workspace_name);
         Ok(())
     }
+
+    /// Resizes the container, as if it was dragged at the edge to a certain point
+    /// on the screen.
+    pub fn resize_container(&mut self, id: Uuid, edge: ResizeEdge, pointer: Point) -> CommandResult {
+        if let Ok(mut lock) = try_lock_action() {
+            if let Some(ref mut action) = *lock {
+                if try!(self.0.lookup(id)).floating() {
+                    self.0.resize_floating(id, edge, pointer, action)
+                } else {
+                    self.0.resize_tiled(id, edge, pointer, action)
+                }
+            } else {
+                // TODO errors
```

Comment:
Well the point is that I return SOME kind of error, I elided this because I wasn't sure how to expand it. One of the issues with returning a generic error with CommandResult, is if I want to expand the Error enum (as I would here, by adding an error for the action already being performed), I would have to change all the enums...it seems to be one of the downsides of Result style error handling, it can be difficult to expand because not all the cases are handled everywhere...wild carding helps, but that isn't always what you want obviously.